lib/handle_callback.ts

```diff
@@ -19,8 +18,9 @@ import {
 import { getAndDeleteOAuthSession, setSiteSession } from "./_kv.ts";
 
 /** Options for {@linkcode handleCallback}. */
-export interface HandleCallbackOptions {
-  /** Overwrites cookie properties set in the response. These must match the
+export interface HandleCallbackOptions<T extends unknown = unknown> {
```

Comment:
Is there a way to strictly type-guard against `null` within the type argument?

Reply:
Does `NonNullable<unknown>` work? https://www.typescriptlang.org/docs/handbook/utility-types.html#nonnullabletype

Reply:
No, because it also guards against `undefined`. We could use `NonNullable<unknown>` if we also don't want the callback returning `undefined`. Do you think there's any benefit to doing that?